src/auto-complete-element.ts

```diff
@@ -12,7 +12,7 @@ export default class AutocompleteElement extends HTMLElement {
 
     // eslint-disable-next-line custom-elements/no-dom-traversal-in-connectedcallback
     const input = this.querySelector('input')
-    const results = document.getElementById(listId)
+    const results = this.querySelector('#'+listId)
```

Comment:
This should probably be `this.ownerDocument.getElementById`, as `this.querySelector` would only allow querying of elements scoped within the element itself. I'd also like to avoid using `querySelector` with string concatenation as it is prone to injection attacks, which isn't the case with `getElementById`.

Reply:
`this.ownerDocument === document`  (tested in chrome), so it wouldn't solve the issue (at least in the case where `auto-complete` is inside a shadowDOM. Could use `this.getRootNode().getElementById` if preferred.

For the scope limiting question becomes: Is having the related elements scattered throughout the document an intended use case? At least the used input element is scoped to `auto-complete`.

Reply:
Typically idrefs can live outside of the element. I think we'd like to continue that expectation.